library/common/src/main/java/com/google/android/exoplayer2/ProducerReferenceTimeBox.java

```diff
@@ -0,0 +1,89 @@
+package com.google.android.exoplayer2;
+
+import androidx.annotation.Nullable;
+import com.google.android.exoplayer2.util.Util;
+
+public final class ProducerReferenceTimeBox {
+  /**
+   * PRFT type related
+   * Use int to align with ExoPlayer practices
+   */
+  public static final int TYPE_ENCODER = 0;
+  public static final int TYPE_CAPTURED = 1;
+
+  /**
+   * NTP related
+   */
+
+  // Baseline NTP time if bit-0=0 is 7-Feb-2036 @ 06:28:16 UTC
+  private static final long msb0baseTime = 2085978496000L;
+
+  // Baseline NTP time if bit-0=1 is 1-Jan-1900 @ 01:00:00 UTC
+  private static final long msb1baseTime = -2208988800000L;
+
+  private final long referenceTrackId;
+  private final long ntpTimestamp;
+  private final long mediaTime;
+  private final int type;
+
+  public ProducerReferenceTimeBox(long referenceTrackId, long ntpTimestamp, long mediaTime, int type) {
+    this.referenceTrackId = referenceTrackId;
+    this.ntpTimestamp = ntpTimestamp;
+    this.mediaTime = mediaTime;
+    this.type = type;
+  }
+
+  /**
+   * Clone from Apache Commons Net NTP TimeStamp getUTCTime() to avoid importing extra dependency.
+   *
+   * @return the number of milliseconds since January 1, 1970, 00:00:00 GMT
+   * represented by this NTP timestamp value.
+   */
+  public long getUTCTime() {
+    final long seconds = (ntpTimestamp >>> 32) & 0xffffffffL;     // high-order 32-bits
+    long fraction = ntpTimestamp & 0xffffffffL;             // low-order 32-bits
+
+    // Use round-off on fractional part to preserve going to lower precision
+    fraction = Math.round(1000D * fraction / 0x100000000L);
+
+    /*
+     * If the most significant bit (MSB) on the seconds field is set we use
+     * a different time base. The following text is a quote from RFC-2030 (SNTP v4):
+     *
+     *  If bit 0 is set, the UTC time is in the range 1968-2036 and UTC time
+     *  is reckoned from 0h 0m 0s UTC on 1 January 1900. If bit 0 is not set,
+     *  the time is in the range 2036-2104 and UTC time is reckoned from
+     *  6h 28m 16s UTC on 7 February 2036.
+     */
+    final long msb = seconds & 0x80000000L;
+    if (msb == 0) {
+      // use base: 7-Feb-2036 @ 06:28:16 UTC
+      return msb0baseTime + (seconds * 1000) + fraction;
+    }
+    // use base: 1-Jan-1900 @ 01:00:00 UTC
+    return msb1baseTime + (seconds * 1000) + fraction;
+  }
+
+  public long getMediaTime() {
+    return mediaTime;
+  }
+
+  public String getType() {
```

Comment:
Do we need to return string?

Reply:
Depends on where to put the "translation". Of course it can return int here and translate them to string for debug overlay under `MyExoPlayer`. I think both imp are fine for current usage.

Reply:
I think it will be more consistent to leave the translation/presentation to debug overlay

Reply:
Sure will move to debug overlay